cli/src/test/java/com/devonfw/tools/ide/context/ProcessContextGitMock.java

```diff
@@ -181,7 +182,8 @@ public ProcessResult run(ProcessMode processMode) {
       }
     }
     this.arguments.clear();
-    return new ProcessResultImpl("git", command.toString(), this.exitCode, this.outs, this.errors);
+    setProcessResult(getExitCode(), this.processResult.getOutputMessages());
+    return this.processResult;
```

Comment:
`setExitCode()` and `getExitCode()` are currently not used in the implementation. The exitcode is set differently in conditions with e.g. `this.exitcode = 0`

What is here the best solution? Is it the right way to use getters and setters inside a class, also for mock classes or can we just assign the attributes with values directly?